src/Language/PureScript/Interactive.hs

```diff
@@ -90,22 +90,23 @@ make ms = do
 -- | Performs a PSCi command
 handleCommand
   :: (MonadReader PSCiConfig m, MonadState PSCiState m, MonadIO m)
-  => (String -> m ())
-  -> m ()
+  => (String -> m ()) -- ^ evaluate JS
```

Comment:
I was also thinking about it. We can implement something like `PSCiT`.

```haskell
type PSCiEvalJS = String -> IO ()
type PSCiReload = IO ()
type PSCiPrint = String -> IO ()
type PSCiFunctions = (PSCiEvalJS, PSCiReload, PSCiPrint)

newtype PSCiT m a = { runPSCiT :: PSCiFunctions -> m a }

class MonadPSCi m where
  psciEvalJS ::  String -> m ()
  psciReload :: m ()
  psciPrint :: String -> m ()

instance MonadIO m => MonadPSCi (PSCiT m) where
  psciEvalJS js = runPSCiT >>> \ (e, _, _) -> liftIO $ e js
  psciReload = runPSCiT >>> \ (_, r, _) -> liftIO r
  psciPrint str = runPSCiT >>> \ (_, _, p) -> liftIO $ p str
```

Does it look good?